### ISSUE TITLE
Max listeners exceeded warning

### ISSUE THREAD

**Issue by gebederry** [IlyaGershman/easy-poll#1]
Hello,
I run the example codes and got this warning after completing 5 pollings.
```shell
(node:1137767) MaxListenersExceededWarning: Possible EventTarget memory leak detected. 11 abort listeners added to [AbortSignal]. Use events.setMaxListeners() to increase limit
    at [kNewListener] (node:internal/event_target:568:17)
    at [kNewListener] (node:internal/abort_controller:241:24)
    at EventTarget.addEventListener (node:internal/event_target:681:23)
    at /opt/apps/demo/easy-poll/node_modules/@ilyagershman/easy-poll/dist/easy-poll.cjs:12:12
    at new Promise (<anonymous>)
    at abortablePromise (/opt/apps/demo/easy-poll/node_modules/@ilyagershman/easy-poll/dist/easy-poll.cjs:6:10)
    at abortableFetch (/opt/apps/demo/easy-poll/node_modules/@ilyagershman/easy-poll/dist/easy-poll.cjs:164:32)
    at _poll (/opt/apps/demo/easy-poll/node_modules/@ilyagershman/easy-poll/dist/easy-poll.cjs:197:28)
```

**Comment by IlyaGershman** [comment]
Thanks for opening this issue @gebederry!

After initial investigation, the warning seems to be quite common, when using abort signals in nodejs. It happens when passing the same signal to different fetch functions.

https://github.com/nodejs/node/issues/46525

I believe the community is waiting for this to be resolved in future nodejs versions

Anyway to be safe, I will run more tests to make sure there's no memory leaks and add setMaxListeners to the example code

**Comment by gebederry** [comment]
> Thanks for opening this issue @gebederry!
> 
> After initial investigation, the warning seems to be quite common, when using abort signals in nodejs. It happens when passing the same signal to different fetch functions.
> 
> [nodejs/node#46525](https://github.com/nodejs/node/issues/46525)
> 
> I believe the community is waiting for this to be resolved in future nodejs versions
> 
> Anyway to be safe, I will run more tests to make sure there's no memory leaks and add setMaxListeners to the example code

Ah, IC. Thank you! Your package has made things really convenient.

**Comment by IlyaGershman** [comment]
Checked this issue more, there's indeed a memory leak when adding event listeners for the abort signal. I will push the fix shortly.

Big thanks @gebederry once again!

**Comment by IlyaGershman** [comment]
The issue was fixed in v3.0.8 🚀
